src/probnum/linops/_linear_operator.py

```diff
@@ -520,6 +709,59 @@ def inv(self) -> "LinearOperator":
 
         return self.__inverse()
 
+    def symmetrize(self) -> LinearOperator:
```

Comment:
By introduction of this method the linear operator `Symmetrize` in `_kronecker.py` should just be removed, I think. Otherwise there are two different ways of achieving the same thing.

Reply:
`Symmetrize` in `_kronecker.py` symmetrizes vectorized matrices used in Kronecker algebra, so it operates "on the arguments" of linear operators, while `linop.symmetrize` symmetrizes the `LinearOperator` itself, so I'd say they serve distinct purposes.

Reply:
They do, but `Symmetrize` is really only used for `SymmetricKronecker` and is arguably confusing in its naming if there is a `symmetrize()` method. My suggestion would be to fold `Symmetrize` into the `SymmetricKronecker` implementation as a (future) refactor. I'm okay with considering this as out of scope for this PR.

Reply:
I agree, let's make a separate PR here :+1